pipes-and-filters/ImageProcessingPipeline/Watermark.cs

```diff
@@ -0,0 +1,46 @@
+using Azure.Storage.Blobs.Models;
+using Azure.Storage.Blobs.Specialized;
+using Microsoft.Azure.Functions.Worker;
+using Microsoft.Extensions.FileProviders;
+using Microsoft.Extensions.Logging;
+using SixLabors.ImageSharp;
+using SixLabors.ImageSharp.Processing;
+
+namespace ImageProcessingPipeline
+{
+    public class Watermark(ILogger<Watermark> logger, IFileProvider files)
+    {
+        private readonly ILogger<Watermark> _logger = logger;
+        private readonly IFileProvider _files = files;
+
+        [Function(nameof(Watermark))]
+        [QueueOutput("pipe-yhrb", Connection = "pipe")]
+        public async Task<string> RunAsync(
+            [QueueTrigger("pipe-fjur", Connection = "pipe")] string imageFilePath,
+            [BlobInput("{QueueTrigger}", Connection = "pipe")] BlockBlobClient imageBlob,
+            CancellationToken cancellationToken)
+        {
+            _logger.LogInformation("Processing image {uri} for watermarking.", imageBlob.Uri);
+
+            // Download image and watermark it
+            using BlobDownloadStreamingResult imageBlobContents = await imageBlob.DownloadStreamingAsync(null, cancellationToken);
+            var image = await Image.LoadAsync(imageBlobContents.Content, cancellationToken);
+            image.Mutate(async i =>
```

Comment:
I had one instance where the watermark was only partially drawn onto the output file. I'm speculating that the ImageSharp mutate doesn't handle the async lambda properly, even though you have an await. I tested moving lines 30-31 before image.Mutate and then removing the async from the image.Mutate(...) call, and it seems to work as expected.

Reply:
Good change. I never ran into it, but the consistency that you received with the change is good enough for me.  Also added the missing cancellationToken usage in this move. You'll find this change also in [`ac2a0bf` (#118)](https://github.com/mspnp/cloud-design-patterns/pull/118/commits/ac2a0bf1d4a2dfadfcd14f629d3a3f26b97a6f50)